db.go

```diff
@@ -203,6 +205,42 @@ func (database *Database) query(w http.ResponseWriter, r *http.Request) {
 	respond(w, http.StatusOK, result)
 }
 
+func (database *Database) getByIds(w http.ResponseWriter, r *http.Request) {
+	conf, auth, err := middleware.Extract(r, true)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	col := ""
+
+	_, r.URL.Path = ShiftPath(r.URL.Path)
+	col, r.URL.Path = ShiftPath(r.URL.Path)
+
+	var body map[string][]string
+	if err := json.NewDecoder(r.Body).Decode(&body); err != nil {
```

Comment:
I'd use the `parseBody` which handle closing the body as well.

Reply:
Fixed